modules/sdk-coin-apt/src/lib/utils.ts

```diff
@@ -191,6 +196,38 @@ export class Utils implements BaseUtils {
   getTxnExpirationTimestamp(): number {
     return Math.floor(Date.now() / 1e3) + SECONDS_PER_WEEK;
   }
+
+  /**
+   * Parses a raw transaction string and returns a `Transaction` object based on its type.
+   *
+   * @param {string} rawTx - The raw transaction string in hex format.
+   * @param {Readonly<StaticsBaseCoin>} coinConfig - The coin configuration object for the transaction.
+   * @returns {Transaction} The transaction object.
+   * @throws {InvalidTransactionError} If the transaction type is invalid or unrecognized.
+   */
+  getTransactionFromRawTx(rawTx: string, coinConfig: Readonly<StaticsBaseCoin>): Transaction {
```

Comment:
We should not define this function in utils.
TransactionBuilderFactory and TransactionBuilder should be the only places where a transaction can be build from.
Otherwise there will be duplicate logic here as well as in TransactionBuilderFactory, and we will always need to maintain the code at both the places.
`Maintenance overhead example: `if a new TransactionType needs to be added in the future, it has to be added here and in `TransactionBuilderFactory`.

Reply:
take reference from [here](https://github.com/BitGo/BitGoJS/blob/master/modules/sdk-coin-apt/src/apt.ts#L157) and use it in the `verifyTransaction()` method to get the required transaction from the transaction builder.